Dockerfile

```diff
@@ -22,78 +22,80 @@
 # 8. add Brew metadata
 
-# Stage 1 - Install dependencies
+# Stage 1 - Build nodejs skeleton
 #@follow_tag(registry.redhat.io/ubi9/nodejs-18:1)
-FROM registry.access.redhat.com/ubi9/nodejs-18:1 AS deps
+FROM registry.access.redhat.com/ubi9/nodejs-18:1 AS skeleton
 USER 0
 
+# Install isolate-vm dependencies
+RUN dnf update -y && \
+  dnf install -y zlib-devel brotli-devel && \
+  dnf clean all
+
 # Env vars
 ENV YARN=./.yarn/releases/yarn-1.22.19.cjs
 ENV REMOTE_SOURCES=.
 ENV REMOTE_SOURCES_DIR=/opt/app-root/src
 
 WORKDIR $REMOTE_SOURCES_DIR/
-COPY $REMOTE_SOURCES $REMOTE_SOURCES_DIR
-RUN chmod +x $REMOTE_SOURCES_DIR/.yarn/releases/yarn-1.22.19.cjs
+COPY $REMOTE_SOURCES/.yarn $REMOTE_SOURCES_DIR/.yarn
+COPY $REMOTE_SOURCES/.yarnrc.yml $REMOTE_SOURCES_DIR/
+RUN chmod +x $REMOTE_SOURCES_DIR/$YARN
 
-# Remove all files except package.json
-RUN find packages -mindepth 2 -maxdepth 2 \! -name "package.json" -exec rm -rf {} \+
+# Stage 2 - Install dependencies
+FROM skeleton AS deps
 
-RUN $YARN install --frozen-lockfile --network-timeout 600000 --ignore-scripts
+COPY $REMOTE_SOURCES/package.json $REMOTE_SOURCES/yarn.lock $REMOTE_SOURCES_DIR
+COPY $REMOTE_SOURCES/packages/app/package.json $REMOTE_SOURCES_DIR/packages/app/package.json
+COPY $REMOTE_SOURCES/packages/backend/package.json $REMOTE_SOURCES_DIR/packages/backend/package.json
 
-# Stage 2 - Build packages
-#@follow_tag(registry.redhat.io/ubi9/nodejs-18:1)
-FROM registry.access.redhat.com/ubi9/nodejs-18:1 AS build
-USER 0
+RUN $YARN install --frozen-lockfile --network-timeout 600000
 
-# Env vars
-ENV YARN=./.yarn/releases/yarn-1.22.19.cjs
-ENV REMOTE_SOURCES=.
-ENV REMOTE_SOURCES_DIR=/opt/app-root/src
+# Stage 3 - Build packages
+FROM deps AS build
 
-WORKDIR $REMOTE_SOURCES_DIR/
 COPY $REMOTE_SOURCES $REMOTE_SOURCES_DIR
-COPY --from=deps $REMOTE_SOURCES_DIR $REMOTE_SOURCES_DIR
-RUN chmod +x $REMOTE_SOURCES_DIR/.yarn/releases/yarn-1.22.19.cjs
+
 RUN git config --global --add safe.directory $REMOTE_SOURCES_DIR/
 RUN rm $REMOTE_SOURCES_DIR/app-config.yaml && mv $REMOTE_SOURCES_DIR/app-config.example.yaml $REMOTE_SOURCES_DIR/app-config.yaml
 
 RUN $YARN build --filter=backend
 
-# Stage 3 - Build the actual backend image and install production dependencies
+# Stage 4 - Build the actual backend image and install production dependencies
+FROM skeleton AS cleanup
+
+# Copy the install dependencies from the build stage and context
+COPY --from=build $REMOTE_SOURCES_DIR/yarn.lock $REMOTE_SOURCES_DIR/package.json $REMOTE_SOURCES_DIR/packages/backend/dist/skeleton.tar.gz $REMOTE_SOURCES_DIR/
+RUN tar xzf skeleton.tar.gz && rm skeleton.tar.gz
+
+# Copy the built packages from the build stage
+COPY --from=build $REMOTE_SOURCES_DIR/packages/backend/dist/bundle.tar.gz $REMOTE_SOURCES_DIR/
+RUN tar xzf $REMOTE_SOURCES_DIR/bundle.tar.gz && rm $REMOTE_SOURCES_DIR/bundle.tar.gz
+
+# Copy app-config files needed in runtime
```

Comment:
see comment above about copying everything from `$REMOTE_SOURCES/` to `$REMOTE_SOURCES_DIR/` in one operation to avoid missing files